src/layout.js

```diff
@@ -163,9 +163,18 @@ export function parseLength(s) {
  * @param {!LengthDef|string} length
  * @return {!LengthDef}
  */
-export function assertLength(length) {
-  assert(/^\d+(\.\d+)?(px|em|rem|vh|vw|vmin|vmax)$/.test(length),
-      'Invalid length value: %s', length);
+export function assertLength(length, allowPercentAsLength) {
```

Comment:
Could you please split this in two methods. We'll leave `assertLength` as is and instead add `assertLengthOrPercent`. `assertLength` is used in a number of places where we don't allow `%`